metacatalog/ext/io/reader.py

```diff
@@ -23,9 +23,15 @@ def read_from_internal_table(entry, datasource, start=None, end=None, **kwargs):
         sql += " AND tstamp <= '%s'" % (dt.strftime(end, '%Y-%m-%d %H:%M:%S'))
 
     # infer table column names order
-    col_sql = 'select * from %s limit 0' % tablename
-    col_names = list(pd.read_sql_query(col_sql, session.bind).columns.values)
-    col_names.remove('entry_id')
+    if datasource.data_name is not None:
+        col_names = datasource.data_name
+    elif entry.variable.column_names is not None:
+        col_names = variable.column_names
+    else:
+        col_sql = 'select * from %s limit 0' % tablename
+        col_names = list(pd.read_sql_query(col_sql, session.bind).columns.values)
+        col_names.remove('entry_id')

```

Comment:
Here use *always* the `datasource.data_name` and fill the correct names in the importer.